pkg/printers/jsonpath_flags.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package printers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+// JSONPathPrintFlags provides default flags necessary for template printing.
+// Given the following flag values, a printer can be requested that knows
+// how to handle printing based on these values.
+type JSONPathPrintFlags struct {
+	// indicates if it is OK to ignore missing keys for rendering
+	// an output template.
+	AllowMissingKeys *bool
+	TemplateArgument *string
+}
+
+// ToPrinter receives an outputFormat and returns a printer capable of
+// handling --template format printing.
+// Returns false if the specified outputFormat does not match a template format.
+func (f *JSONPathPrintFlags) ToPrinter(outputFormat string) (ResourcePrinter, bool, error) {
+	if (f.TemplateArgument == nil || len(*f.TemplateArgument) == 0) && len(outputFormat) == 0 {
+		return nil, false, fmt.Errorf("missing --template value")
+	}
+
+	templateArg := ""
+
+	// templates are logically optional for specifying a format.
+	// this allows a user to specify a template format value
+	// as --output=jsonpath=
+	templateFormats := map[string]bool{
+		"jsonpath":      true,
+		"jsonpath-file": true,
+	}
+
+	if f.TemplateArgument == nil || len(*f.TemplateArgument) == 0 {
+		for format := range templateFormats {
+			format = format + "="
+			if strings.HasPrefix(outputFormat, format) {
+				templateArg = outputFormat[len(format):]
```

Comment:
This is messed up, I'd suggest you create two separate variables:
1. for the template format (this will be `jsonpath` or `jsonpath-file`)
2. for the template value/argument (this will be `{.metadata.name}`)

Reply:
Thanks, done